app/scripts/modules/core/pipeline/config/copyStage/copyStage.modal.controller.ts

```diff
@@ -0,0 +1,96 @@
+import {flatten} from 'lodash';
+import {module} from 'angular';
+
+import {API_SERVICE, Api} from 'core/api/api.service';
+import {Application} from 'core/application/application.model';
+import {COPY_STAGE_CARD_COMPONENT} from './copyStageCard.component';
+import {IPipeline} from 'core/domain/IPipeline';
+import {IStage} from 'core/domain/IStage';
+
+import './copyStage.modal.less';
+
+interface IStageWrapper {
+  pipeline: string;
+  stage: IStage;
+}
+
+class CopyStageModalCtrl implements ng.IComponentController {
+  viewState = { loading: true, error: false };
+  applications: Application[];
+  stages: IStageWrapper[];
+  selectedStage: IStageWrapper;
+
+  static get $inject() { return ['$q', 'API', 'application', 'applicationReader', '$uibModalInstance']; }
+
+  constructor (public $q: ng.IQService,
+               public API: Api,
+               public application: Application,
+               public applicationReader: any,
+               public $uibModalInstance: any) { }
```

Comment:
most of these can be private, right? `application` probably needs to be public, but not the others